webui/react/src/hooks/useSettings.ts

```diff
@@ -258,27 +252,32 @@ const useSettings = <T>(config: SettingsConfig<T>): UseSettingsReturn<T> => {
 
       const array = settingsArray ?? Object.keys(config.settings);
 
-      stateOb.update((s) => {
-        const news = s.get(config.storagePath);
-        array.forEach((setting) => {
-          let defaultSetting: SettingsConfigProp<T[Extract<keyof T, string>]> | undefined =
-            undefined;
+      stateOb.update((prev) =>
+        prev.withMutations((map) => {
+          const oldSettings = map.get(config.storagePath) ?? {};
+          const newSettings = { ...oldSettings };
```

Comment:
nit: looks like `oldSettings` is not used other than assigning
 
```suggestion
          const newSettings = map.get(config.storagePath) ?? {};
```

Reply:
Ah yeah, I can probably get rid of `oldSettings` but would still need to make sure that it's a new object to ensure the change is detected.